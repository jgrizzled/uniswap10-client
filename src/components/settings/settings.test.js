import Settings from '.';

const settings = {
  currency: 'usd',
  liquidityWeight: 0.5,
  rebalancePeriod: 30
};
const component = <Settings settings={settings} setSettings={jest.fn()} />;

// smoke test
it('renders without crashing', async () => {
  const div = document.createElement('div');
  ReactDOM.render(<Providers>{component}</Providers>, div);
  ReactDOM.unmountComponentAtNode(div);
});

// shallow snapshot test
it('renders as expected', async () => {
  const wrapper = Enzyme.mount(component, { wrappingComponent: Providers });
  expect(wrapper).toMatchSnapshot();
});
