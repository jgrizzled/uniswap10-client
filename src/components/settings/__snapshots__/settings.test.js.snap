// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<Settings
  setSettings={[MockFunction]}
  settings={
    Object {
      "currency": "usd",
      "liquidityWeight": "0.5",
      "rebalancePeriod": "30",
    }
  }
>
  <styled.div>
    <div
      className="sc-AxjAm qfvsW"
    >
      <styled.h2>
        <h2
          className="sc-AxirZ jDYAnS"
        >
          Settings
        </h2>
      </styled.h2>
      <styled.div>
        <div
          className="sc-AxiKw fAKbNN"
        >
          <form
            onSubmit={[Function]}
          >
            <fieldset>
              <legend>
                Currency: 
              </legend>
              <label>
                <input
                  checked={false}
                  name="currency"
                  onChange={[Function]}
                  type="radio"
                  value="eth"
                />
                ETH 
              </label>
              <label>
                <input
                  checked={true}
                  name="currency"
                  onChange={[Function]}
                  type="radio"
                  value="usd"
                />
                USD 
              </label>
            </fieldset>
            <fieldset>
              <legend>
                Rebalance Period: 
              </legend>
              <label>
                <input
                  checked={false}
                  name="rebalancePeriod"
                  onChange={[Function]}
                  type="radio"
                  value="7"
                />
                Weekly 
              </label>
              <label>
                <input
                  checked={true}
                  name="rebalancePeriod"
                  onChange={[Function]}
                  type="radio"
                  value="30"
                />
                Monthly 
              </label>
              <label>
                <input
                  checked={false}
                  name="rebalancePeriod"
                  onChange={[Function]}
                  type="radio"
                  value="90"
                />
                Quarterly 
              </label>
            </fieldset>
            <fieldset>
              <legend>
                Asset Weighting: 
              </legend>
              <label>
                <input
                  checked={false}
                  name="liquidityWeight"
                  onChange={[Function]}
                  type="radio"
                  value="1"
                />
                Liquidity 
              </label>
              <label>
                <input
                  checked={true}
                  name="liquidityWeight"
                  onChange={[Function]}
                  type="radio"
                  value="0.5"
                />
                Liquidity and volume 
              </label>
              <label>
                <input
                  checked={false}
                  name="liquidityWeight"
                  onChange={[Function]}
                  type="radio"
                  value="0"
                />
                Volume 
              </label>
            </fieldset>
            <button
              type="submit"
            >
              Apply
            </button>
          </form>
        </div>
      </styled.div>
    </div>
  </styled.div>
</Settings>
`;
